df.viewscreen.xml

```diff
@@ -476,25 +476,105 @@
+                <int32_t name='unk_9'/>
+                <int32_t name='unk_10'/>
+                <int32_t name='year'/>
+                <int32_t name='year_tick'/>
+                <int32_t name='unk_12'/> -1
+                <int32_t name='army_controller2' ref-target='army_controller'/>
+                <int32_t name='histfig' ref-target='historical_figure'/>
+                <int32_t name='unk_14'/> -1
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <int32_t/>
+                <stl-vector name='squads' type-name='int32_t' ref-target='squad'/>
+                <stl-vector name='messengers' type-name='int32_t' ref-target='occupation'/>
+
+                <int32_t/>
+                <int32_t/>
+
+                FIXME: this points to different objects depending on mission type
+                       size 128 for raid, 8 for artifact and person recovery
+                <pointer name='details'> 
+                    recovery: artifact_id for artifact recovery mission, histfig_id for person recovery
+                    raid: 4 - raid  6 - pillage  0 - raze  3 - demand tribute  2 - occupy  1 - take over  5 - explore
+                    <int32_t name='target_id'/> 
+                    <int32_t name='unk_2'/>
+
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t name='raid_flags'/> 128 - steal livestock  64 - loot other items  32 - take important treasures  16 - release other prisoners  8 - free captives belonging to your civ.  2 - one-time tribute  4 - ongoing tribute  512 - demand surrender
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                    <int32_t/>
+                </pointer>
+
+                <int32_t name='unk_15'/>
+                <int32_t/>
+
+            </pointer>            
+        </stl-vector>
         <int32_t name='mission_idx'/>
         <stl-vector name='squads' pointer-type='squad'/>
-        <stl-vector name='squad_unk' type-name='int32_t'/>
+        <stl-vector name='squads_travelling' type-name='int32_t'/>
```

Comment:
Unit IDs? If so, you can use a `ref-target='unit'` attribute here (you don't need to expand the vector contents like you did with item types)

Reply:
Oops, squads aren't units - I meant for this to go with messengers_travelling below. Maybe ref-target=squad?

Reply:
_travelling are just vectors of zeroes and ones showing corresponding squad/messenger availability.